app-website-4/ProductsApi/Controllers/AuthController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using ProductsApi.Auth;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace ProductsApi.Controllers
+{
+    [Route("api/[controller]")]
+    public class AuthController : Controller
+    {
+        private readonly TokenProvider _tokenProvider;
+
+        public AuthController(TokenProvider tokenProvider)
+        {
+            _tokenProvider = tokenProvider;
+        }
+
+        [HttpPost("token")]
+        public IActionResult Token()
+        {
+            var header = Request.Headers["Authorization"];
```

Comment:
You could lift this logic up to `Startup.cs` by using a package like this: https://github.com/blowdart/idunno.Authentication

You'd define that as your 'BasicAuthentication' scheme and then add an `Authorize` attribute on this controller action, specifying that same auth scheme.

The rest of your API routes could then use an `Authorize` attribute which specifies the JWT authentication type.

That way, you can centralise all the logic to do with auth in a single place (`Startup.cs`), rather than doing it in this controller.

Reply:
I'll take a look.